src/rest_framework_jwt/management/commands/obtain_token.py

```diff
@@ -0,0 +1,39 @@
+import sys
+
+from django.core.management.base import BaseCommand, CommandError
+
+
+class Command(BaseCommand):
+    help = 'Create new jwt token for the requested user'
+
+    def add_arguments(self, parser):
+        parser.add_argument('filter', type=str, help='ex: pk=7 or username=apollo or email=foo@bar')
+
+    def _parse_filter(self, filter_value):
+        sep_index = str(filter_value).find('=')
+        if sep_index == -1:
+            raise ValueError("'{}' is not a valid filter".format(filter_value))
+        lookup_field, lookup_value = filter_value[0:sep_index], filter_value[sep_index+1:]
+        return {
+            lookup_field: lookup_value
+        }
+
+    def handle(self, *args, **options):
+        try:
+            from rest_framework_jwt.settings import api_settings
+            from django.contrib.auth import get_user_model
+            user = get_user_model().objects.filter(**self._parse_filter(options['filter']))
+            if len(user) == 0:
+                raise CommandError("There is no user with the given filter")
+            if len(user) > 1:
+                raise CommandError("There are more than one user with the given filter")
+            user = user[0]
```

Comment:
Could you use [`User.objects.filter().first()`](https://docs.djangoproject.com/en/3.0/ref/models/querysets/#first)?

Reply:
Hmm, I don't know why I didn't use `first()`, I'll change it

Reply:
`_parse_filter` method will return a dictionary describing a condition, and there is no guarantee that applying this condition on the queryset will result in only one object, and I didn't want to implicitly decide to generate the token for the "first match".
that's why I didn't use the `first()` method.

Reply:
You're saying you want to filter against any `User` field, not just the `pk` and the `username` as stated in the documentation? Why? Is there a use-case for filtering against the rest of the fields and then narrowing down to the most eligible choice?

Reply:
I didn't have a specific use-case in mind, but still I'd like to keep the developer's options open. if they have a custom User model, that model could have many uique fields which we don't know here, I wanted to let the developer choose which field they want to use for filtering.
Although now it sounds a bit over-engineered considering that it is a management command to use during development.
I'll remove extra logic

Reply:
In that case, isn't it easier to support only PK lookups? After all, that's the only common denominator across implementations, is it not?